### PR TITLE
Angular 9 update

### PR DESCRIPTION
**Pull request by zainniazi**
- Angular 9 updates to packages
- Support for latest Ivy engine

### CODE REVIEW

**Comment by adamkleingit** [comment]
Hi @zainniazi thanks for the PR.
Can you say what is the `static forRoot(): ModuleWithProviders<TreeModule> {` needed for?
Was there a warning/error, or it's just for better typings?
Thanks
BTW, there's another PR open for Angular 9:
https://github.com/500tech/angular-tree-component/pull/815
We need to merge the best of both somehow

**Comment by slavabogov** [comment]
@adamkleingit Regarding `ModuleWithProviders<TreeModule>`: https://angular.io/guide/migration-module-with-providers

> Libraries should add the generic type to any usages of the ModuleWithProviders type.

File: tsconfig.json [hunk 1]
```diff
@@ -19,6 +19,6 @@
   "buildOnSave": false,
   "angularCompilerOptions": {
     "skipTemplateCodegen": true,
-    "enableIvy": false
+    "enableIvy": true
```
Comment (mistic100) [inline tsconfig.json:22]: This is discouraged by Angular CLI itself as it makes librairies not compatible with previous versions, it is totally fine to leave `false`

**Comment by mistic100** [comment]
@adamkleingit the other PR aims to changed how the lib is builded to respect Angular Package Format, it is not mandatory for Angular 9 compatibility.

This PR on the other hand is the strict minimum to make it working. It would be really helpful if you could reviewe and publish it.

File: package.json [hunk 2]
```diff
@@ -62,53 +62,56 @@
...
   ],
   "peerDependencies": {
-    "core-js": "^2.4.1"
+    "core-js": "^3.6.4"
   },
   "dependencies": {
-    "lodash": "^4.17.11",
-    "mobx": "^4.15.1"
+    "lodash": "^4.17.15",
+    "mobx": "^5.15.4",
```
Comment (mistic100) [inline package.json:69]: Please don't, mobx 5 is not compatible with IE 11

**Pull request closed without merging**